client/app/bundles/course/assessment/pages/AssessmentStatistics/StudentMarksPerQuestionTable.tsx

```diff
@@ -158,53 +159,42 @@ const StudentMarksPerQuestionTable: FC<Props> = (props) => {
     );
   };
 
-  // the customised sorting for grades to ensure null always is less than any non-null grade
-  const sortNullableGrade = (
-    grade1: number | null,
-    grade2: number | null,
-  ): number => {
-    if (!grade1 && !grade2) {
-      return 0;
-    }
-    if (!grade1) {
-      return -1;
-    }
-    if (!grade2) {
-      return 1;
-    }
-    return grade1 - grade2;
-  };
-
   const answerColumns: ColumnTemplate<MainSubmissionInfo>[] = Array.from(
     { length: assessment?.questionCount ?? 0 },
     (_, index) => {
       return {
         searchProps: {
-          getValue: (datum) => datum.answers?.[index]?.grade?.toString() ?? '',
+          getValue: (datum) =>
+            datum.answers?.[index]?.grade?.toString() ?? undefined,
         },
         title: t(translations.questionIndex, { index: index + 1 }),
         cell: (datum): ReactNode => {
-          return typeof datum.answers?.[index].grade === 'number'
-            ? renderAnswerGradeClickableCell(index, datum)
-            : null;
+          return typeof datum.answers?.[index].grade === 'number' ? (
+            renderAnswerGradeClickableCell(index, datum)
+          ) : (
+            <div />
+          );
         },
         sortable: true,
         csvDownloadable: true,
         className: 'text-right',
         sortProps: {
-          sort: (datum1, datum2): number => {
-            return sortNullableGrade(
-              datum1.answers?.[index]?.grade ?? null,
-              datum2.answers?.[index]?.grade ?? null,
-            );
-          },
+          undefinedPriority: 'last',
         },
       };
     },
   );
 
-  const jointGroupsName = (datum: MainSubmissionInfo): string =>
-    datum.groups ? datum.groups.map((g) => g.name).join(', ') : '';
+  const jointGroupsName = (datum: MainSubmissionInfo): string | undefined => {
```

Comment:
Why make it so complicated? Your previous code can simply add a `.sort()` to the chain  
`datum.groups ? datum.groups.map((g) => g.name).sort().join(', ') : '';`

Since it's a string type, you can also remove `sortProps.undefinedPriority: 'last'` from the column config for the groups column.

Same goes for `StudentAttemptCountTable.tsx`